CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+cmake_minimum_required(VERSION 3.0)
+
+# Better behavior of VS generator.
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+set_property(GLOBAL PROPERTY PREDEFINED_TARGETS_FOLDER "CMake")
+
+# Allow to specify VERSION in project()
+cmake_policy(SET CMP0048 NEW)
+
+set(ZLIBCOMPLETE_VERSION 1.0.5)
+
+project(zlibcomplete VERSION ${ZLIBCOMPLETE_VERSION})
+
+add_definitions(-DZLIBCOMPLETE_VERSION="${ZLIBCOMPLETE_VERSION}"
+                -DZLIBCOMPLETE_GITHUB="https://github.com/HexwellC/zlibcomplete")
+
+# Generate compilation database for YouCompleteMe and stuff like clang-tidy.
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Allow some tweaks in build process without changing CMakeLists.
+option(ZLIBCOMPLETE_WERROR  "Build with -Werror flag."                 OFF)
+option(ZLIBCOMPLETE_WSTRICT "Build with really strict warning flags."  OFF)
+option(ZLIBCOMPLETE_RELEASE "Build with optimizations"                 ON)
+
+# Check if used compiler supports warning flags we use.
+if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") OR
+   ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") OR
+   ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang"))
+    if (ZLIBCOMPLETE_WSTRICT)
+        set(WARNING_FLAGS ${WARNING_FLAGS} -Wall -Wextra -pedantic -Weffc++
+                -Wcast-align -Wcast-qual -Wno-ctor-dtor-privacy
+                -Wdisabled-optimization -Wformat=2 -Winit-self
+                -Wmissing-declarations -Wmissing-include-dirs -Wold-style-cast
+                -Woverloaded-virtual -Wredundant-decls -Wshadow
+                -Wsign-conversion -Wsign-promo -Wstrict-overflow=5 -Wswitch
+                -Wundef -Wno-unused -Wnon-virtual-dtor -Wreorder -Wdeprecated
+                -Wno-float-equal
+                )
+    endif()
+    if(ZLIBCOMPLETE_WERROR)
+        set(WARNING_FLAGS ${WARNING_FLAGS} -Werror)
+    endif()
+else()
+    message(WARNING "Using WERROR/WSTRICT with non-GCC like compiler is not supported. Ignoring.")
+endif()
+
+# Documentation (Doxygen)
+
+find_package(Doxygen)
+option(ZLIBCOMPLETE_DOCS    "Generate HTML documentation (Doxygen)"    ${DOXYGEN_FOUND})
+if(ZLIBCOMPLETE_DOCS)
+    if (NOT DOXYGEN_FOUND)
+        message(FATAL_ERROR "Doxygen is required to build documentation.")
+        # Don't have to put everything else in else branch because FATAL_ERROR aborts configuration.
+    endif()
+
+    configure_file(${CMAKE_SOURCE_DIR}/Doxyfile ${CMAKE_BINARY_DIR}/Doxyfile @ONLY)
```

Comment:
Meaningless, zlibcomplete doesn't uses CMake variables in Doxyfile, thus no `configure_file` required.